### DISCUSSION TITLE
AO and Backfaces

### DISCUSSION THREAD

**Discussion by aVersionOfReality** [bnpr/Malt#567] [General]
I am using some double sided meshes instead of backface culling. It seems that Malt is automatically inverting Normals on backfaces. And for Normals from Geometry Nodes, I invert them with Is Backfacing:
![image](https://github.com/bnpr/Malt/assets/36803956/aea66f46-e46d-41a2-a5cd-805ca18b3070)

But I am having problems with Ambient Occlusion.
Here is a shirt model. The Normals on the collar section point inwards because it folds over, as seen on the left. On the right, it has been Solidified, so they all point in the correct direction.

![image](https://github.com/bnpr/Malt/assets/36803956/f7f6a6f4-f147-4a56-8590-3b0be6fadfe0)

And here is the AO. It is not happy on the backfaces.

![image](https://github.com/bnpr/Malt/assets/36803956/ff404a68-e432-48a5-a169-15a46e8cf1b5)

My question is, is there something I can invert on Backfaces to get the AO behaving properly? I don't really understand how AO interacts with Normals.

In the case of this shirt, I can just invert the Normals on the collar. But that won't always be an option on more complex meshes.

**Comment by pragma37** [comment]
Actually, the issue was that normals were not being inverted for ambient occlusion.
I've just committed a fix: 08949f94ea2763a439fd847e9a6b9a1d9ed4c73a

**Reply by aVersionOfReality** [reply]
Oh an actual bug, not just me misunderstanding something. Thanks for the fix!
